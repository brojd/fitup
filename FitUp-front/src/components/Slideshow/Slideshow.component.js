import React, { PropTypes } from 'react';
import Image from '../Image/Image.component';
import classNames from 'classnames';
import styles from './Slideshow.stylesheet.css';

class Slideshow extends React.Component {
  constructor() {
    super();
    this.positions = [
      { top: '20px', left: '-100px' },
      { top: '70px', right: '-100px' },
      { top: '200px', left: '-130px' },
      { top: '120px', left: '10px' }
    ];
  }
  textPosition() {
    return this.positions[this.props.currentTextPosition];
  }
  render() {
    return (
      <div className={classNames(styles.Slideshow, 'uk-width-10-10')}>
        <div className={classNames(styles.slideSection, 'uk-width-4-10 uk-align-center')}>
          {
            this.props.slides.map(
              (slide, i) =>
                <div className={classNames(styles.slide, 'uk-align-center')}
                     key={slide.id}>
                  <Image src={this.props.slides[i].imgSrc}
                         alt={`image ${i}`}
                         width={600}
                         className={this.props.slides[i].visible===true ?
                            classNames('uk-align-center', styles.slide__img, 'fadeIn') :
                            classNames('uk-align-center', styles.slide__img, styles.hide)} />
                  <h2 className={this.props.slides[i].visible===true ?
                        classNames('uk-align-center', styles.slide__text, 'zoomInUp') :
                        classNames('uk-align-center', styles.slide__text, styles.hide)}
                      style={this.textPosition()} >
                    {this.props.slides[i].text}
                  </h2>
                </div>
            )
          }
        </div>
      </div>
    );
  }
}

Slideshow.propTypes = {
  slides: PropTypes.arrayOf(Object),
  currentTextPosition: PropTypes.number
};

export default Slideshow;
