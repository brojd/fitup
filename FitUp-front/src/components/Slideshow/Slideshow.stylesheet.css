.Slideshow {
  background: #000000;
  height: 450px;
  overflow: hidden;
  box-shadow: 0 2px 5px 0 #350000; }

.slideSection {
  height: inherit;
  position: relative; }

.slide {
  height: inherit;
  position: absolute; }

.slide__text {
  color: #FFFFFF;
  position: absolute;
  font-family: Century Gothic, CenturyGothic, AppleGothic, sans-serif !important;
  font-size: 25px;
  text-shadow: 0px 0px 6px #000000, 1px 1px 4px #FF1B1B, -1px -1px 4px #FF1B1B; }

.hide {
  opacity: 0; }

/*# sourceMappingURL=Slideshow.stylesheet.css.map */
