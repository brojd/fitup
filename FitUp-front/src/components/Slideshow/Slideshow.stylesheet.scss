@import "../../../sass_variables/colors";

.Slideshow {
  background: $black;
  height: 450px;
  overflow: hidden;
  box-shadow: 0 2px 5px 0 $temptress;
}

.slideSection {
  height: inherit;
  position: relative;
}

.slide {
  height: inherit;
  position: absolute;
}

.slide__text {
  color: $white;
  position: absolute;
  font-family: Century Gothic, CenturyGothic, AppleGothic, sans-serif !important;
  font-size: 25px;
  text-shadow: 0px 0px 6px $black, 1px 1px 4px $torch_red, -1px -1px 4px $torch_red;
}

.hide {
  opacity: 0;
}
