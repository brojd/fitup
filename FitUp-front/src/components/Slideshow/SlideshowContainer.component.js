import React, { PropTypes } from 'react';
import Slideshow from './Slideshow.component';

class SlideshowContainer extends React.Component {
  
  constructor(props) {
    super(props);
    this._changeSlide = this._changeSlide.bind(this);
    this._updateTextPosition = this._updateTextPosition.bind(this);
    let slides = this.props.slides.slice();
    this.state = {
      currentSlide: 0,
      slides: slides,
      currentTextPosition: 0
    };
  }
  
  _changeSlide() {
    let currentIndex = this.state.currentSlide;
    if (currentIndex === (this.state.slides.length - 1)) {
      currentIndex = 0;
      this.setState({
        currentSlide: 0,
        slides: (this.state.slides.map(
          (n, i) => Object.assign(n, { visible: (currentIndex === i) })
        ))
      });
    } else {
      currentIndex += 1;
      this.setState({
        currentSlide: this.state.currentSlide + 1,
        slides: (this.state.slides.map(
          (n, i) => Object.assign(n, {
            visible: (currentIndex === i)
          })
        ))
      });
    }
  }
  
  _updateTextPosition() {
    let lastIndex = this.state.slides.length - 1;
    let currentIndex = this.state.currentTextPosition;
    this.setState({ currentTextPosition: (currentIndex === lastIndex) ? 0 : (currentIndex + 1)});
  }
  
  componentDidMount() {
    this._interval = setInterval(() => {
      this._changeSlide();
      this._updateTextPosition();
    }, 3500);
  }
  
  componentWillUnmount() {
    clearInterval(this._interval);
  }
  
  render() {
    return (
      <Slideshow slides={this.state.slides} currentTextPosition={this.state.currentTextPosition}/>
    );
  }
}

SlideshowContainer.propTypes = {
  className: PropTypes.string,
  slides: PropTypes.arrayOf(Object)
};

export default SlideshowContainer;
