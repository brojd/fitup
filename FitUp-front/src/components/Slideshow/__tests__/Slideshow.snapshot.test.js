import React from 'react';
import Slideshow from '../Slideshow.component';
import renderer from 'react-test-renderer';

test('Slideshow renders correctly', () => {
  const slides = [
    {
      id: 1,
      imgSrc: 'http://www.noupe.com/wp-content/uploads/2012/10/pixabay-beach-49877.jpg',
      text: 'All fitness classes on one website',
      visible: true
    },
    {
      id: 2,
      imgSrc: 'http://www.noupe.com/wp-content/uploads/2012/10/pixabay-beach-49879.jpg',
      text: 'Filter the schedule easily',
      visible: false
    },
    {
      id: 3,
      imgSrc: 'http://sm4u.pl/wp-content/uploads/2014/04/Pixabay-Free-photos.png',
      text: 'Review all trainers from Wroclaw',
      visible: false
    },
    {
      id: 4,
      imgSrc: 'https://cdn.pixabay.com/blog/post/2013/03/23/23-02-28-603_640.jpg',
      text: 'Save your time',
      visible: false
    }
  ];
  const component = renderer.create(
    <Slideshow slides={slides} />
  );
  let tree = component.toJSON();
  
  expect(tree).toMatchSnapshot();
});
