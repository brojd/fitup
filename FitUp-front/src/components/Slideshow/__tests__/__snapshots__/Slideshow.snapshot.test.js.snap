exports[`test Slideshow renders correctly 1`] = `
<div
  className="Slideshow uk-width-10-10">
  <div
    className="slideSection uk-width-4-10 uk-align-center">
    <div
      className="slide uk-align-center">
      <img
        alt="image 0"
        className="uk-align-center slide__img fadeIn"
        height={150}
        onClick={undefined}
        src="http://www.noupe.com/wp-content/uploads/2012/10/pixabay-beach-49877.jpg"
        width={600} />
      <h2
        className="uk-align-center slide__text zoomInUp"
        style={undefined}>
        All fitness classes on one website
      </h2>
    </div>
    <div
      className="slide uk-align-center">
      <img
        alt="image 1"
        className="uk-align-center slide__img hide"
        height={150}
        onClick={undefined}
        src="http://www.noupe.com/wp-content/uploads/2012/10/pixabay-beach-49879.jpg"
        width={600} />
      <h2
        className="uk-align-center slide__text hide"
        style={undefined}>
        Filter the schedule easily
      </h2>
    </div>
    <div
      className="slide uk-align-center">
      <img
        alt="image 2"
        className="uk-align-center slide__img hide"
        height={150}
        onClick={undefined}
        src="http://sm4u.pl/wp-content/uploads/2014/04/Pixabay-Free-photos.png"
        width={600} />
      <h2
        className="uk-align-center slide__text hide"
        style={undefined}>
        Review all trainers from Wroclaw
      </h2>
    </div>
    <div
      className="slide uk-align-center">
      <img
        alt="image 3"
        className="uk-align-center slide__img hide"
        height={150}
        onClick={undefined}
        src="https://cdn.pixabay.com/blog/post/2013/03/23/23-02-28-603_640.jpg"
        width={600} />
      <h2
        className="uk-align-center slide__text hide"
        style={undefined}>
        Save your time
      </h2>
    </div>
  </div>
</div>
`;
